libnmstate/nm/bond.py

```diff
@@ -45,20 +83,38 @@ def is_bond_type_id(type_id):
 
 def get_bond_info(nm_device):
     slaves = get_slaves(nm_device)
-    options = get_options(nm_device)
+    options = _get_options(nm_device)
     if slaves or options:
         return {"slaves": slaves, "options": options}
     else:
         return {}
 
 
-def get_options(nm_device):
-    con = connection.ConnectionProfile()
-    con.import_by_device(nm_device)
-    if con.profile:
-        bond_settings = con.profile.get_setting_bond()
-        return bond_settings.props.options
-    return {}
+def _get_options(nm_device):
+    ifname = nm_device.get_iface()
+    if not ifname:
+        return {}
+
+    options = {}
+    for sysfs_file in glob.iglob(f"/sys/class/net/{ifname}/bonding/*"):
+        option = os.path.basename(sysfs_file)
+        if option in NM_SUPPORTED_BOND_OPTIONS:
+            options[option] = _read_sysfs_file(f"{sysfs_file}")
```

Comment:
This is generating an options dict with elements of `NM_SUPPORTED_BOND_OPTIONS` on the left side and the values on the right side. Is that okay? Shouldn't we have our API constants on the left instead of `NM_SUPPORTED_BOND_OPTIONS`?

Sorry if I am missing something here.

Reply:
Don't provide a full list of supported bond options. That's is maintained by NM.